webapp/security/models.py

```diff
@@ -166,6 +166,7 @@ class Status(Base):
         )
     )
     description = Column(String)
+    component = Column(Enum("main", "universe", name="components"))
```

Comment:
We spoke about this defaulting to "main" if it's not set, could you do that? I can't remember exactly how this is done in SQLAlchemy, but it shouldn't be too hard.

Reply:
I believe you should also be adding "server_default=main" here.

Reply:
Well I've been playing around a bit with inserting models to see if they default properly, but I can't get them to work, even though I added `server_default` and `default`.

``` python3
component = Column(Enum("main", "universe", name="components", default="main", server_default="main"))
```

``` bash
In : from webapp.security.models import Release, Status, CVE, Package

In : db_session.add(Status(status="Released", release=db_session.query(Release).first(), cve=db_session.query(CVE).first(), package=db_session.query(Package).first(), description="dfgd"))
In : db_session.commit()

IntegrityError: (psycopg2.errors.NotNullViolation) null value in column "component" violates not-null constraint
DETAIL:  Failing row contains (CVE-2002-2439, gcc-4.1, warty, released, dfgd, null).

[SQL: INSERT INTO status (cve_id, package_name, release_codename, status, description, component) VALUES (%(cve_id)s, %(package_name)s, %(release_codename)s, %(status)s, %(description)s, %(component)s)]
[parameters: {'cve_id': 'CVE-2002-2439', 'package_name': 'gcc-4.1', 'release_codename': 'warty', 'status': 'released', 'description': 'dfgd', 'component': None}]
(Background on this error at: http://sqlalche.me/e/13/gkpj)
```

Reply:
I added `server_default="main",` in the model too.
I set the `server_default` in the `Column` not inside the `Enum`.

I ran some tests where I provide no component to make sure it works and it did default it do `main`, it's just a bit difficult to tell if the default is set at a model level or a database level.

I only had to change my code to not take the `component` for granted:
```python
            if "component" in status_data:
                status.component = status_data["component"]
```